File: backend/caesar.py
from typing import Optional

DEFAULT_SHIFT = 3

def _shift_char(c: str, shift: int) -> str:
    if 'a' <= c <= 'z':
        return chr((ord(c) - ord('a') + shift) % 26 + ord('a'))
    if 'A' <= c <= 'Z':
        return chr((ord(c) - ord('A') + shift) % 26 + ord('A'))
    return c

def encrypt(text: str, shift: Optional[int] = None) -> str:
    """Encrypt using a Caesar cipher (simple shift)."""
    if shift is None:
        shift = DEFAULT_SHIFT
    return ''.join(_shift_char(c, shift) for c in text)

def decrypt(text: str, shift: Optional[int] = None) -> str:
    """Decrypt a Caesar-ciphered string."""
    if shift is None:
        shift = DEFAULT_SHIFT
    return ''.join(_shift_char(c, -shift) for c in text)

File: frontend/src/utils/caesar.js
export const DEFAULT_SHIFT = 3

function shiftChar(c, shift){
  if(c >= 'a' && c <= 'z'){
    return String.fromCharCode((c.charCodeAt(0) - 'a'.charCodeAt(0) + shift + 26) % 26 + 'a'.charCodeAt(0))
  }
  if(c >= 'A' && c <= 'Z'){
    return String.fromCharCode((c.charCodeAt(0) - 'A'.charCodeAt(0) + shift + 26) % 26 + 'A'.charCodeAt(0))
  }
  return c
}

export function decrypt(text, shift = DEFAULT_SHIFT){
  // reverse the Caesar shift
  return String(text).split('').map(c => shiftChar(c, -shift)).join('')
}

export function encrypt(text, shift = DEFAULT_SHIFT){
  return String(text).split('').map(c => shiftChar(c, shift)).join('')
}
